subsystems/motor/SensorMotor.java

```diff
@@ -6,108 +6,153 @@
 		motionController.reset();
 	}
-	
+
 	public void setInputRange(double minimum, double maximum) {}
-	
+
 	public void enablePID() {
 		isMotionControlEnabled = true;
 		motionController.enable();
 	}
-	
+
 	public void disablePID() {
 		isMotionControlEnabled = false;
 		motionController.disable();
 	}
-	
+
 	/**
 	 * Set the position of a sensor motor
 	 *
 	 * @param position
 	 * @throws InvalidSensorException
 	 */
 	public void setPositionSafely(double position) throws InvalidSensorException {
+		PIDSourceType previousSensorSourceType = motionController.getSensorSourceType();
+		motionController.setSensorSourceType(PIDSourceType.kDisplacement);
 		motionController.setSetpoint(position);
 		motionController.enable();
 		double speed = motionController.getSafely();
+		motionController.setSensorSourceType(previousSensorSourceType);
 		LogKitten.v(getName() + " set to position " + position + " at speed " + speed);
 		super.set(speed);
 	}
-	
+
 	/**
 	 * Set the position of a sensor motor
-	 * 
+	 *
 	 * @param position
 	 * @warning this does not indicate sensor failure
 	 */
 	public void setPosition(double position) {
+		PIDSourceType previousSensorSourceType = motionController.getSensorSourceType();
+		motionController.setSensorSourceType(PIDSourceType.kDisplacement);
 		motionController.setSetpoint(position);
 		motionController.enable();
 		double speed = motionController.get();
+		motionController.setSensorSourceType(previousSensorSourceType);
 		LogKitten.v(getName() + " set to position " + position + " at speed " + speed);
 		super.set(speed);
 	}
-	
+
 	public boolean onTarget() {
 		return motionController.onTarget();
 	}
-	
+
+	/**
+	 * Set the velocity of a sensor motor
+	 *
+	 * @param distancePerSecond
+	 * @throws InvalidSensorException
+	 */
+	public void setVelocitySafely(double distancePerSecond) throws InvalidSensorException {
+		PIDSourceType previousSensorSourceType = motionController.getSensorSourceType();
+		motionController.setSensorSourceType(PIDSourceType.kRate);
+		motionController.setSetpoint(distancePerSecond);
+		motionController.enable();
+		double speed = motionController.getSafely();
+		motionController.setSensorSourceType(previousSensorSourceType);
+		LogKitten.v(getName() + " set to velocity " + distancePerSecond + " at speed " + speed);
+		super.set(speed);
+	}
+
+	/**
+	 * Set the velocity of a sensor motor
+	 *
+	 * @param distancePerSecond
+	 * @warning this does not indicate sensor failure
+	 */
+	public void setVelocity(double distancePerSecond) {
+		PIDSourceType previousSensorSourceType = motionController.getSensorSourceType();
+		motionController.setSensorSourceType(PIDSourceType.kRate);
+		motionController.setSetpoint(distancePerSecond);
+		motionController.enable();
+		double speed = motionController.get();
+		motionController.setSensorSourceType(previousSensorSourceType);
+		LogKitten.v(getName() + " set to velocity " + distancePerSecond + " at speed " + speed);
+		super.set(speed);
+	}
```

Comment:
This will work, and I think there could be good reasons for doing it, but I would like to note the possibility of having this function try/catch the setVelocitySafely function. setPositionSafely is in a similar state.
Is there a reason for either?